contracts/sophia.md

```diff
@@ -149,32 +149,27 @@ type transaction = SpendTx(spend_tx)
 
 #### Contract primitives
 
-The block-chain environment available to a contract is defined by three record types
-`call`, `self` and `chain`:
-
-```
-type call  = {caller : address,
-              value  : uint,
-              ...}
-type self  = {creator : address,
-              address : address,
-              balance : uint,
-              ...}
-type chain = {height    : uint,
-              timestamp : uint,
-              ...}
-```
-A contract has access to variables of the same names, so that
-
-- `call.caller` is the address of the entity (possibly another contract)
+The block-chain environment available to a contract is defined in three name spaces
+`Contract`, `Call`, and `Chain`:
+
+- `Contract.creator` is the address of the entity that signed the contract creation
+  transaction.
+- `Contract.address` is the address of the contract account.
+- `Contract.balance` is the amount of coins currently in the contract account.
```

Comment:
I expect this is the balance in the chain state passed as input to the contract call (as other txs placed before this tx in the same block may have changed the chain state).